kubeflow/core/tensorboard.libsonnet

```diff
@@ -0,0 +1,252 @@
+
+    // true Whether or not to use https for S3 connections
+    s3UseHttps: "true",
+
+    // Whether or not to verify https certificates for S3 connections
+    s3VerifySsl: "true",
+
+    // URL for your s3-compatible endpoint.
+    s3Endpoint: "http://s3.us-west-1.amazonaws.com,",
+  } + $.params,
+
+
+  components:: {
+
+    all::
+      // TODO(jlewi): It would be better to structure s3 as a mixin.
+      // As an example it would be great to allow S3 and GCS parameters
+      // to be enabled simultaneously. This should be doable because
+      // each entails adding a set of environment variables and volumes
+      // to the containers. These volumes/environment variables shouldn't
+      // overlap so there's no reason we shouldn't be able to just add
+      // both modifications to the base container.
+      // I think we want to restructure things as mixins so they can just
+      // be added.
+      if $.params.s3Enable then
+        [
+          $.s3parts.tb,
+          $.s3parts.tfDeployment,
+        ]
+      else if $.params.cloud == "gcp" then
+        [
+          $.gcpParts.tb,
+          $.gcpParts.tfDeployment,
+        ]
+      else
+        [
+          $.parts.tb,
+          $.parts.tfDeployment,
+        ],
+  }.all,
+
+  parts:: {
+    // We define the containers one level beneath parts because combined with jsonnet late binding
+    // this makes it easy for users to override specific bits of the container.
+    tbContainer:: {
+      name: $.params.name,
+      image: $.params.defaultTbImage,
+      imagePullPolicy: "IfNotPresent",
+      args: [
+        $.params.logDir,
+        "--port=9000"
+      ],
+      ports: [
+        {
+          containerPort: 9000,
+        },
+      ],
+      
+      resources: {
+        requests: {
+          memory: "1Gi",
+          cpu: "1",
+        },
+        limits: {
+          memory: "4Gi",
+          cpu: "4",
+        },
+      },
+    },  // tbContainer
+
+    tfDeployment: {
+      apiVersion: "extensions/v1beta1",
+      kind: "Deployment",
+      metadata: {
+        name: $.params.name,
+        namespace: $.params.namespace,
+        labels: $.params.labels,
+      },
+      spec: {
+        template: {
+          metadata: {
+            labels: $.params.labels,
+          },
+          spec: {
+            containers: [
+              $.parts.tbContainer
+            ],
+            
+          },
+        },
+      },
+    },  // tfDeployment
+
+    tb: {
+      apiVersion: "v1",
+      kind: "Service",
+      metadata: {
+        labels: $.params.labels,
+        name: $.params.name,
+        namespace: $.params.namespace,
+        annotations: {
+          "getambassador.io/config":
+            std.join("\n", [
+              "---",
+              "apiVersion: ambassador/v0",
+              "kind:  Mapping",
+              "name: tb-mapping-" + $.params.name + "-get",
+              "prefix: " + $.params.name + "/",
```

Comment:
Shouldn't the prefix be: "/tensorboard/" + $params.name + "/"

I think you need to start with a "/" and we also want a common prefix e.g. "tensorboard"